roles/tendrl-server/tasks/tendrl-api.yml

```diff
@@ -13,21 +13,41 @@
   notify:
     - restart tendrl-api
 
-- name: Configure tendrl-api etcd.yml user_name
+- name: Configure tendrl-api etcd.yml user_name to empty string
   replace:
     dest: /etc/tendrl/etcd.yml
     regexp: "^ +:user_name:.*"
     replace: "  :user_name: ''"
   notify:
     - restart tendrl-api
+  when: etcd_authentication == False
 
-- name: Configure tendrl-api etcd.yml password
+- name: Configure tendrl-api etcd.yml user_name
+  replace:
+    dest: /etc/tendrl/etcd.yml
+    regexp: "^ +:user_name:.*"
+    replace: "  :user_name: 'root'"
+  notify:
+    - restart tendrl-api
+  when: etcd_authentication == True
+
+- name: Configure tendrl-api etcd.yml password to empty string
   replace:
     dest: /etc/tendrl/etcd.yml
     regexp: "^ +:password:.*"
     replace: "  :password: ''"
   notify:
     - restart tendrl-api
+  when: etcd_authentication == False
```

Comment:
If etcd authentication is already enabled, we should not disable it in tendrl (the same apply for other tendrl components).

Reply:
Oh right, the problem here is that the default config of api looks like this:

```
:development:
  :base_key: ''
  :host: '127.0.0.1'
  :port: 2379
  :user_name: 'username'
  :password: 'password'
```

and so I need to either:

* replace the default credentials with empty string when etcd auth is disabled
* fill out the credentials when etcd auth is enabled

I can't leave it alone ...

Reply:
Created issue for the tendrl-api here: https://github.com/Tendrl/api/issues/294

Reply:
Addressed in 39210f7847f53e8eb4e02ba60a581af860fe6e53. Until https://github.com/Tendrl/api/issues/294 is not fixed, disabling etcd auth will not work.